ecl/hql/hqlutil.cpp

```diff
@@ -4794,9 +4794,10 @@ bool BitfieldPacker::checkSpaceAvailable(unsigned & thisBitOffset, unsigned & th
 {
     bool fitted = true;
     thisBits = type->getBitSize();
-    if (thisBits > bitsRemaining)
+    ITypeInfo * storeType = type->queryChildType();
+    if ((thisBits > bitsRemaining) || (activeType && storeType != activeType))
```

Comment:
You are correct about the condition.  I think what I meant to write was

if ((thisBits > bitsRemaining) || !activeType || (storeType != activeType))

Strictly speaking the test is redundant, but I would still tend to keep the test there to explicitly show it is checking for that case.